aiohttp/client.py

```diff
@@ -611,34 +613,40 @@ async def _request(
                             resp.release()
 
                         try:
-                            parsed_url = URL(
+                            parsed_redirect_url = URL(
                                 r_url, encoded=not self._requote_redirect_url
                             )
-
                         except ValueError as e:
-                            raise InvalidURL(r_url) from e
+                            raise InvalidRedirectURL(r_url) from e
 
-                        scheme = parsed_url.scheme
+                        scheme = parsed_redirect_url.scheme
                         if scheme not in ("http", "https", ""):
                             resp.close()
-                            raise ValueError("Can redirect only to http or https")
+                            raise InvalidRedirectURL(
+                                r_url, "Can redirect only to http or https"
+                            )
                         elif not scheme:
-                            parsed_url = url.join(parsed_url)
+                            parsed_redirect_url = url.join(parsed_redirect_url)
 
                         is_same_host_https_redirect = (
-                            url.host == parsed_url.host
-                            and parsed_url.scheme == "https"
+                            url.host == parsed_redirect_url.host
+                            and parsed_redirect_url.scheme == "https"
                             and url.scheme == "http"
                         )
 
+                        try:
+                            redirect_origin = parsed_redirect_url.origin()
+                        except ValueError as e:
+                            raise InvalidRedirectURL(parsed_redirect_url) from e
```

Comment:
Why not have a description so that this case is distinguishable for the one a few checks above?
```suggestion
                        except ValueError as origin_val_err:
                            raise InvalidRedirectURL(
                                parsed_redirect_url,
                                "Invalid redirect URL origin",
                            ) from origin_val_err
```

Reply:
I probably wanted to keep exception the same, with just class updated but adding description does not break anything :+1: